Constant folding for lifted graph

zhxchen17 opened the pull request:
Summary:
Current implementation for lifted graph takes a dict of [constant name: constant value]. And the constant value is used to run_node and excute the constant graph to get the folded values and then create new getattr nodes for folded values.

We don't have constant values for lifted graph during model compilation on MTIA. I think it is more general to allow the constant folding pass to just take the constant names only to produce the constant graph and represent the folded nodes as placeholders to make it consistent with lifted graph. Additionally, this mimic the real situation on Sigmoid, where Sigmoid executes the constant graph, get the folded values and set the folded values to the main graph. This diff is to update the pass to work with a list of constant names.

Test Plan:
```
buck run mode/opt caffe2/test:test_export -- -r split_const_gm
```

Differential Revision: D62144791


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

zhxchen17 commented:
clarification: I'm not the author of this diff, but I clicked the "Export" button on Phabricator.

eellison commented:
many test failures

eellison reviewed (commented):
many test failures - also cc @muchulee8 who added this part

zhxchen17 commented:
@pytorchbot rebase

SherlockNoMad commented:
Reviewed via diff.

trieuat committed:
Constant folding for lifted graph (#135060)

Summary:

Current implementation for lifted graph takes a dict of [constant name: constant value]. And the constant value is used to run_node and excute the constant graph to get the folded values and then create new getattr nodes for folded values.

However, we don't have constant values for lifted graph during model compilation on MTIA. I think it makes the pass more general by allowing it to just take the constant names only to produce the constant graph and represent the folded nodes as placeholders to make it consistent with lifted graph. Additionally, this mimic the real situation in Sigmoid, where Sigmoid executes the constant graph, get the folded values and set the folded values to the main graph. This diff is to update the pass to work with a list of constant names.

Test Plan:
```
buck run mode/opt caffe2/test:test_export -- -r split_const_gm
```

Reviewed By: muchulee8, SherlockNoMad

Differential Revision: D62144791

wdvr commented:
@pytorchbot merge -i
